MekHQ/src/mekhq/campaign/personnel/SpecialAbility.java

```diff
@@ -161,14 +161,39 @@ public boolean isEligible(Person p) {
         }
         return true;
     }
-    
+
+    public boolean isEligible(boolean isClanner, Skills skills, PersonnelOptions options) {
+        for(SkillPrereq sp : prereqSkills) {
+            if(!sp.qualifies(skills)) {
+                return false;
+            }
+        }
+        for(String ability : prereqAbilities) {
+            //TODO: will this work for choice options like weapon specialist?
```

Comment:
It'll probably detect that the person has the category of SPA, but not the specific sub-type.

Reply:
This is all existing code and not my TODO's, but good to know it won't work for a sub type.